### ISSUE TITLE
question about paper

### ISSUE THREAD

**Issue by diaodeyi** [YotamNitzan/ID-disentanglement#9]
Hi，thanks for your beautiful work about ID-disentanglement， i have some question as follows：
1. why only fixed the Eid and G, the Eaddr and Eind shouldn't be fixed?
2. what the relationship between the LG(adv) and LG(non-adv) ? i mean the order of optimization.the paper says they are optimized separately , but no matter optimize which first ,  it will change the result  after optimize the other . I think there should be some relationship between them.
3. just like question 2, LD(adv) use z, this z is not the concatenation of the Eattr and Eid ，right？Because the GT has only one W per time,i means we doesn't have the W's GT.

**Comment by YotamNitzan** [comment]
Hi @diaodeyi,
Pleased to hear you liked our work :)

1. First, E_lnd is also fixed. Only the E_attr and M are trained. Next, the fixed networks are fixed because we have off-the-shelf networks to use for their purpose. For example, extracting identity or face landmarks. On the other hand, the attribute network is tasked with an ill-defined task to extract "everything that is not identity". No such pretrained network exists and therefore it must be trained as a part of the entire pipeline.
2. LG(adv) is optimized only in half of all iterations, while LG(non-adv) is always trained. Please look at trainer.py for more details.  
3. z is indeed the concatenation you described, and as appear in equation (1) in the paper. You are also correct that we don't have a w ground-truth, but an adversarial loss doesn't require one.

**Comment by diaodeyi** [comment]
Thank you so much, as for quesntion 3,   you want to train the MLP with the Discriminator ，so the adversarial loss is for MLP, i mean we have to train the MLP first, so that you have correct W from z, and then you can train other network. So what‘s the training sequence between the LG(adv) and LG(non-adv)？ (i think  LG(adv)  is first, and not alternative training）

**Comment by diaodeyi** [comment]
Sorry, As your paper said "and the latent 𝑤 vectors are used as "real" samples for training 𝐷W"
and the LD(adv) indeed  need the W as the GT for input. 
so what do you mean  "but an adversarial loss doesn't require one." in question 3?

**Comment by diaodeyi** [comment]
> Sorry, As your paper said "and the latent 𝑤 vectors are used as "real" samples for training 𝐷W"
> and the LD(adv) indeed need the W as the GT for input.
> so what do you mean "but an adversarial loss doesn't require one." in question 3?

i refer the LD(adv) not LG(adv)

**Comment by YotamNitzan** [comment]
Hi @diaodeyi,
I'll try to explain the entire process.

First, we use the pretrained StyleGAN to simply generate a dataset of W values. This part doesn't depend on our method and on any of the losses you mentioned. Simply forwarding many Gaussian vectors through StyleGAN to obtain a dataset of Ws.

Next, in our training, we have a different mapping network, M, which maps z = [E_id,E_attr] to the W space. Now, to actually map into W space, we use an adversarial loss. The "fake" is our prediction from M and the "real" is the dataset of W we generated in the first step. 

What I meant by "adversarial loss doesn't require a ground truth", is that, we don't know what is the correct output (or ground truth) for M(z). We only apply an adversarial loss that forces M(z) to lie within the W distribution.
